lib/std/process.zig

```diff
@@ -676,13 +676,18 @@ pub const ArgIteratorWindows = struct {
 
     /// `cmd_line_w` *must* be a WTF16-LE-encoded string.
     ///
-    /// The iterator makes a copy of `cmd_line_w` converted WTF-8 and keeps it; it does *not* take
-    /// ownership of `cmd_line_w`.
+    /// The iterator stores and uses `cmd_line_w`, so its memory must be valid for
+    /// at least as long as the returned ArgIteratorWindows.
     pub fn init(allocator: Allocator, cmd_line_w: [*:0]const u16) InitError!ArgIteratorWindows {
-        const cmd_line = try unicode.wtf16LeToWtf8Alloc(allocator, mem.sliceTo(cmd_line_w, 0));
-        errdefer allocator.free(cmd_line);
-
-        const buffer = try allocator.alloc(u8, cmd_line.len + 1);
+        const cmd_line = mem.sliceTo(cmd_line_w, 0);
+        const wtf8_len = unicode.calcWtf8Len(cmd_line);
+
+        // This buffer must be large enough to contain contiguous NUL-terminated slices
+        // of each argument. For arguments past the first one, space for the NUL-terminator
+        // is guaranteed due to the necessary whitespace between arugments. However, we need
+        // one extra byte to guarantee enough room for the NUL terminator if the command line
+        // ends up being exactly 1 argument long with no quotes, etc.
+        const buffer = try allocator.alloc(u8, wtf8_len + 1);
```

Comment:
Note: This is not new behavior, this comment just clarifies what's going on with the size of this buffer (as I understand it, anyway).